### PR TITLE
Undo deleted/completed Items

### PR DESCRIPTION
**Pull request by dlloydgeorge**
User can undo deleted/completed items.

### CODE REVIEW

File: BLT/BLT/Focus Mode View/FocusTimer.swift [hunk 1]
```diff
@@ -19,6 +19,8 @@ class FocusTimer {
 	var secs: Int {
 		return Int(cdt) % 60
 	}
+    ///amount of time that the timer has been running for
+    var timeStudied: TimeInterval
```
Comment (EKKOING) [inline BLT/BLT/Focus Mode View/FocusTimer.swift]: This is a very inefficient way of doing this.

Comment (EKKOING) [reply BLT/BLT/Focus Mode View/FocusTimer.swift]: This also does not fit with the description of the PR at all.

File: BLT/BLT/UserData.swift [hunk 2]
```diff
@@ -56,6 +56,12 @@ class UserData: Codable {
 			saveUserData()
 		}
 	}
+    
+    var focusTime: Double = 0 {
```
Comment (EKKOING) [inline BLT/BLT/UserData.swift]: This should be calculated by the app dynamically it cannot be a variable that we just keep adding to.

**Review by EKKOING — CHANGES_REQUESTED** [review]
I think we are gonna have to scrap most of this as there is no way to use a single variable to dynamically calculate the Focus Hours over a given time period.

File: BLT/BLT/Focus Mode View/FocusTimer.swift (hunk 1 above)
Comment (dlloydgeorge) [reply BLT/BLT/Focus Mode View/FocusTimer.swift]: Yeah I added some extra stuff and forgot about it. The time studied is simply the time studied during a single focus session. That then gets added to the variable focusTime in userData

**Review by EKKOING — APPROVED** [review]
Fixed all issues found

**Pull request merged**